moz_telemetry/sandboxes/heka/analysis/moz_telemetry_new_doctype_monitor.lua

```diff
@@ -0,0 +1,140 @@
+
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+# Mozilla Telemetry New DocType Monitor
+
+Monitors the telemetry stream for new document types and alerts if 
+they meet a volume threshold.
+
+## Sample Configuration
+```lua
+filename = "moz_telemetry_new_doctype_monitor.lua"
+message_matcher = "Type == 'telemetry'"
+ticker_interval = 60
+preserve_data = true
+threshold = 100
+
+alert = {
+  disabled = false,
+  prefix = false,
+  threshold = 90,  -- one alert every 90 minutes
+  modules = {
+    email = {recipients = {"amiyaguchi@mozilla.com"}},
+  }
+}
+```
+--]]
+
+_PRESERVATION_VERSION = 0
+
+local alert = require "heka.alert"
+local table = require "table"
+
+-- Helper function to generate a set
+function Set(list)
+    local set = {}
+    for _, l in ipairs(list) do set[l] = true end
+    return set
+end
+
+-- Maintained in the puppet repository: https://github.com/mozilla-services/puppet-config/blob/master/pipeline/yaml/app/pipeline.yaml#L78
+known_doctypes = Set{
```

Comment:
You will want to rethink the creation/modification/preservation of this table: 
- To update this table you would have to bump the preservation version (dumping the history) and create a new package every time
- Ideally you would move the definitive list to the cfg and the discovered list would by dynamic and preserved

Reply:
Updating the list would probably involve moving everything from the discovered list to the known list, which would be worthy of a preservation version bump.